[PULL REQUEST] Feature/blowing snow

jaegle opened the pull request:
Updates to GEOS-Chem v12.7.0 to include blowing snow from sea salt aerosol. 
Submitted by Lyatt Jaegle and Jiayue Huang to GC support team June 2020. Please email jaegle@uw.edu with any questions.

These updates add a new source of sea salt aerosol (SSA) from the wind mobilization of saline snow on sea ice. The blowing snow SSA source follows the parameterization of Yang et al. (2008, 2010) as implemented in GEOS-Chem by Huang and Jaegle (2017) and Huang et al. (2018). It also includes a source of sea salt bromide (BrSALA and BrSALC) from the blowing snow SSA, assuming a surface snow bromide enrichement factor of 5 relative to sea water (sea salt Br–  is emitted assuming a ratio of 10.55 10-3 kg Br per kg of dry blowing snow SSA emitted) as described in Huang et al. (2020).

The default assumptions about the blowing snow source are in the HEMCO_Config file and include:
1) the salinity of surface snow on first year sea ice (FYI) in the NH (0.1 practical salinity units, psu) and SH (0.03 psu)
2) the salinity of surface snow on multi year sea ice (MYI) in the NH (0.05 psu) and SH (0.015)
3) the surface snow age: 3 days in the NH and 1.5 days in the SH
4) the number of SSA particles produced per snowflake (5).

The daily fraction of multi year sea ice is obtained from remotely sensed observations of sea ice motion and sea ice extent for the Arctic (Tschudi et al., 2019). For the Antarctic, the multi year sea ice extent is based on the minimum MERRA-2 sea ice extent of the previous summer.

These changes also include two updates to the SSA emissions from the open ocean. We have added a suppression of open ocean SSA emissions over cold waters with SST<5 deg C as described in Huang and Jaegle (2017). This reduces SSA emissions during summertime by a factor of 2 over cold waters. This is implemented as a flag in HEMCO_Config.rc. Also in prior versions of GEOS-Chem the SSA ocean emissions were only calculated in open ocean gridboxes
as determined by the function HCO_Landtype based on LWI (land water ice) flags and (ALB) Albedo. This lead to no SSA emissions over high latitudes during winter night and also to uneven emissions in coastal areas. The LWI flags have been superseded by the MERRA-2 and
GEOS-FP fields FRLAND, FROCEAN, FRSEAICE. In the updated version we now use FROCEAN and FRSEAICE to determine whether the gridbox is open ocean or sea ice, and we scale the emissions based on the fraction of the gridbox covered by these respective fractions.

References:
  Huang, J. and Jaeglé, L.: Wintertime enhancements of sea salt aerosol in polar regions consistent with a sea ice source from blowing snow, Atmospheric Chem. Phys., 17(5), 3699–3712, doi:10.5194/acp-17-3699-2017, 2017.
  Huang, J., Jaeglé, L. and Shah, V.: Using CALIOP to constrain blowing snow emissions of sea salt aerosols over Arctic and Antarctic sea ice, Atmospheric Chem. Phys., 18(22), 16253–16269, doi:https://doi.org/670 10.5194/acp-18-16253-2018, 2018.
  Huang, J., Jaeglé, L., Chen, Q., Alexander, B., Sherwen, T., Evans, M. J., Theys, N., and Choi, S.: Evaluating the impact of blowing snow sea salt aerosol on springtime BrO and O3 in the Arctic, Atmos. Chem. Phys. Discuss., https://doi.org/10.5194/acp-2019-1094, in press, 2020.
   Tschudi, M., W. N. Meier, J. S. Stewart, C. Fowler, and J. Maslanik.  "EASE-Grid Sea Ice Age, Version 4." NASA National Snow and Ice Data Center Distributed Active Archive Center. doi: https://doi.org/10.5067/UTAV7490FEPB., 2019.
  Yang, X., Pyle, J. A. and Cox, R. A.: Sea salt aerosol production and bromine release: Role of snow on sea ice, Geophys. Res. Lett., 35(16), doi:10.1029/2008GL034536, 2008.
  Yang, X., Pyle, J. A., Cox, R. A., Theys, N. and Van Roozendael, M.: Snow-sourced bromine and its implications for polar tropospheric ozone, Atmospheric Chem. Phys., 10(16), 7763–7773, doi:10.5194/acp-10-7763-2010, 2010.

Modifications have been made to 6 files:
```
 HEMCO/Extensions/hcox_seasalt_mod.F90    | 516 ++++++++++++++++++++++-
 HEMCO/Interfaces/hcoi_standalone_mod.F90 |  33 +-
 Headers/state_met_mod.F90                |  26 ++
 GeosCore/hcoi_gc_main_mod.F90            |  26 ++
 HEMCO/Extensions/hcox_state_mod.F90      |  10 +
 GeosCore/flexgrid_read_mod.F90           |   9 +-
```
Changes to HEMCO_Config.rc Extensions (* indicate new):
```
107     SeaSalt                : on    SALA/SALC/Br2/BrSALA/BrSALC/MOPO/MOPI
    --> SALA lower radius      :       0.01
    --> SALA upper radius      :       0.5
    --> SALC lower radius      :       0.5
    --> SALC upper radius      :       8.0
*   --> Reduce SS cold water   :       true
*   --> Blowing Snow SS        :       true
*   --> NH FYI snow salinity   :       0.1
*   --> NH MYI snow salinity   :       0.05
*   --> SH FYI snow salinity   :       0.03
*   --> SH MYI snow salinity   :       0.015
*   --> NH snow age            :       3.0
*   --> SH snow age            :       1.5
*   --> N per snowflake        :       5.0
    --> Emit Br2               :       false
    --> Br2 scaling            :       1.0
    --> Model sea salt Br-     :       true
    --> Br- mass ratio         :       2.11e-3


#=================================================================
# Sea salt emissions (Extension 107)
#=================================================================
(((SeaSalt
107 MODIS_CHLR $ROOT/MODIS_CHLR/v2019-11/MODIS.CHLRv.V5.generic.025x025.$YYYY.nc MODIS 2005-2014/1-12/1/0 C xy 1 * - 1 1
107 MULTISEAICE $ROOT/MULTI_ICE/v2020-04/multiyearice.merra2.05x0625.$YYYY.nc GMAO_2D__FRSEAICE   1984-2017/1-12/1-31/0 C xy 1 * - 1 1
)))SeaSalt
```
The files in $ROOT/MULTI_ICE/v2020-04 indicate the daily fraction of multi year sea ice. These files were generated from  remotely sensed observations of sea ice motion and sea ice extent for the Arctic (Tschudi et al., 2019). For the Antarctic, the multi year sea ice extent is based on the minimum MERRA-2 sea ice extent of the previous summer.
[HEMCO_Config.rc.txt](https://github.com/geoschem/geos-chem/files/4772635/HEMCO_Config.rc.txt)

jaegle commented:
As input for the blowing snow emissions code I have some HEMCO files with multi-year sea ice fraction for 1984-2018. They total to about 10G. How should I share them with the GC support team?

msulprizio commented:
Thanks @jaegle! I've added this item to the [Model Development Priorities](http://wiki.seas.harvard.edu/geos-chem/index.php/GEOS-Chem_model_development_priorities#Ready_to_go_in) page under "ready to go in" so that it may be considered for inclusion in the GEOS-Chem 13 release series. 

You may send us the data files via an ftp site, Dropbox, or any equivalent file transfer system available to you.

jaegle commented:
The multi-year sea ice files are available on the anonymous ftp site:
ftp.atmos.washington.edu
cd jaegle/MULTI_ICE

msulprizio commented:
Thanks @jaegle! I downloaded the files successfully. We'll keep you posted on bringing these updates into the standard model.

yantosca commented:
Hi Lyatt,

Thanks for preparing the files.  I noticed a few things in the netCDF file structure that will need to be changed so that these files can be read in properly by GCHP.  (GCHP has stricter I/O requirements than GC-Classic.)

1. The time values in all of the files are referenced from 1985.  This will cause problems with GCHP, which always expects the first time in the file to be zero.  The time:units string in each file will have to be changed from:
```
time:units = "hours since 1985-1-1 00:00:0.0" ;
```
to e.g. for 1985:
```
time:units = "hours since 1985-01-01 00:00:00" ;
time = 0, 24, 48, 72, ...
```
e.g. for 1986:
```
time:units = "hours since 1986-01-01 00:00:00" ;
time = 0, 24, 48, 72, ...
```
etc.
You can also use days since instead of hours since but that's up to you.

2.  Could you change the variable name from: `GMAO_2D__FRSEAICE` to `FRSEAICE`  That would be more straightforward.

3. [Use the nc_chunk.pl script to deflate and chunk the files](http://wiki.seas.harvard.edu/geos-chem/index.php/Working_with_netCDF_data_files#Chunking_and_deflating_a_netCDF_file_to_improve_I.2FO).  GCHP requires that the file chunking be (1,NY,NX).  The basic deflation level (1) can really reduce the file size, which makes for easier storage.

@lizziel: Is there anything else that I've forgotten?

jaegle commented:
Hi Bob,
Thanks for noting these issues on the multiyear sea ice files. I have updated them to fix the time, variable name and used nc_chunk.pl to reduce file size. 

The new multi-year sea ice files are available on the anonymous ftp site:
ftp.atmos.washington.edu
cd jaegle/MULTI_ICE

msulprizio commented:
Thanks Lyatt!  I downloaded the updated files.

lizziel commented on HEMCO/Extensions/hcox_seasalt_mod.F90:
```diff
@@ -290,23 +383,170 @@ SUBROUTINE HCOX_SeaSalt_Run( am_I_Root, ExtState, HcoState, RC )
+                     * ( QSNOWICE_FYI * SUM( Inst%F_DN_N_FYI(:,N) ) + &
+                         QSNOWICE_MYI * SUM( Inst%F_DN_N_MYI(:,N) ) ) * DDSNOW
+              ENDIF
+              ! add bromine blowing snow 
+              IF ( Inst%CalcBr2 ) THEN
+                IF ( HcoState%Grid%YEDGE%Val(I,J) .lt. 0 ) THEN
+                SSA_Br2 =  SSA_Br2 + HcoState%TS_EMIS * A_M2 &
+                        * (QSNOWICE_FYI * SUM( Inst%F_DI_S_FYI(:,N) ) + &
+                           QSNOWICE_MYI * SUM( Inst%F_DI_S_MYI(:,N) ) ) * DDSNOW &
+                        * 0.00223d0 * 0.7d0 / 2.0d0
+                ELSE
+                SSA_Br2 =  SSA_Br2 + HcoState%TS_EMIS * A_M2 &
+                        * (QSNOWICE_FYI * SUM( Inst%F_DI_N_FYI(:,N) ) + &
+                           QSNOWICE_MYI * SUM( Inst%F_DI_N_MYI(:,N) ) ) * DDSNOW &
+                        * 0.00223d0 * 0.7d0 / 2.0d0
+                ENDIF
+              ENDIF
+          ENDIF
```
The bromine option in the sea salt extension was retired in 12.9. To accomodate that change I excluded this section in the blowing snow update in GEOS-Chem 13. @jaegle, can you confirm that this is okay? Are there other places that need editing as well? Also, are you okay with the bromine option being retired in the first place?

jaegle replied:
Yes, you can remove the CalcBr2 option. 

A few more places that need editing:

1) the variable AGE should be included in the PRIVATE statement in hcox_seasalt_mod.F90 (it wasn't in the patch I submitted)

2) I noticed that when using HEMCO standalone to calculate blowing snow emissions, surface pressure was set at 101325.0 Pa. I use the variable HcoState%Grid%PSFC%Val( I, J ):
         ! Surface pressure at grid box (I,J). Convert from [Pa] to [hPa]
         PRESS = HcoState%Grid%PSFC%Val( I, J ) /100d0
should a difference variable be used?

lizziel replied:
1. I made that change. Thanks for noticing that.
2. Yes, using that variable should be fine. What you saw in 12.7 may be a bug that has since been fixed where PS was not used to set pressure in the absence of PSFC listed as an input.

lizziel commented:
Hi @jaegle, I am working on bringing your updates into the standard model. There are conflicts in this PR due to differences between GEOS-Chem 13 and 12.7.0 so I am instead manually bringing them on a new branch. I will make a new PR for this feature request using that branch for us to merge in and will link to it here when I am done.

I have a few questions about the updates as I am putting them in.

1. The Br option in the sea salt extension was retired in 12.9. This was an update from Xuan Wang who wrote: 

> I would also suggest to completely retire the SeaSalt Br2 option in HEMCO. This is an option when we do not have heterogeneous bromine chemistry on sea salt. It is not useful since Johan Schmidt’s update was implemented 4 years ago.

 I added a comment to the code in this PR about this. Please see that comment which has questions.

2. Could you provide a README for the data you submitted? A good example of a data README is [here](http://ftp.as.harvard.edu/gcgrid/data/ExtData/HEMCO/CEDS/v2021-06/README). Let me know if you have any questions.

Thank you!

lizziel commented on HEMCO/Extensions/hcox_seasalt_mod.F90:
```diff
@@ -290,23 +383,170 @@ SUBROUTINE HCOX_SeaSalt_Run( am_I_Root, ExtState, HcoState, RC )
-       SCALE = SCALE * Inst%WindScale
+       SCALE = SCALE * Inst%WindScale * FROPEN
+
+       ! There is no need for this conditions as FROPEN=0 if there is
+       ! no open ocean. Also this lead to zeroing of SSA emissions over
+       ! polar night regions 
+       !inhibit emission if not ocean grid box
+       !IF ( HCO_LANDTYPE( ExtState%WLI%Arr%Val(I,J), &
+       !     ExtState%ALBD%Arr%Val(I,J) ) /= 0 ) &
+       !     SCALE = 0d0
```
Since this is commented out does it need to be included?

jaegle replied:
Yes, this can be removed. Note that using the following condition:
IF ( HCO_LANDTYPE( ExtState%WLI%Arr%Val(I,J), &
                                    ExtState%ALBD%Arr%Val(I,J) ) /= 0 )
to determine whether a grid box has open ocean, is present in other parts of HEMCO (such as hcox_seaflux_mod.F90 among others). The ALBD field in the MERRA-2 A1 file is set to a large number (>0.8) for all regions that are in polar night - I assume that this is because it is used/calculated in the radiation part of the MERRA-2 code. As a result, the HCO_LandType function (or any use of the condition ALBD>0.695 or ALBD>0.4) erroneously assumes that open ocean regions that are in polar night are covered by sea ice. Thus no sea salt emissions (or air-sea exchange) occurs in these boxes. This should be corrected. 

One simple fix is to replace the use of ALBD in HCO_LANDTYPE with sea ice fraction (FRSEAICE) and more generally to replace the use of ALBD with FRSEAICE to determine whether an ocean box is covered by sea ice or not.

lizziel replied:
Thanks! Could you create a github issue with this bug report?

lizziel commented:
There are now two new PRs to replace this one:
HEMCO: https://github.com/geoschem/HEMCO/pull/89
GEOS-Chem: https://github.com/geoschem/geos-chem/pull/777

@jaegle, I am also running 1-month benchmarks and will have a comparison to show shortly for validation. If you could also take a quick look over the code changes in each PR that would be very helpful.

lizziel commented:
Validation for the changes will be here later today:
[http://ftp.as.harvard.edu/gcgrid/geos-chem/validation/BlowingSnow](http://ftp.as.harvard.edu/gcgrid/geos-chem/validation/BlowingSnow)

msulprizio commented:
> Validation for the changes will be here later today:
> http://ftp.as.harvard.edu/gcgrid/geos-chem/validation/BlowingSnow

Differences in SeaSalt emissions are summarized below. See also [these difference plots.](http://ftp.as.harvard.edu/gcgrid/geos-chem/validation/BlowingSnow/GCC_version_comparison/Emissions/Natural_Emissions.pdf)

```
###################################################################################
### Emissions totals for inventory SeaSalt [Tg]                                 ###
### Ref = 13.2.0_dev; Dev = 13.2.0_dev+blowingSnow                              ###
###################################################################################
                                    Ref                 Dev     Dev - Ref    % diff
SeaSalt Br2        :           0.000000            0.000000      0.000000       nan
SeaSalt BrSALA     :           0.009512            0.011070      0.001558    16.383
SeaSalt BrSALC     :           0.575690            0.559263     -0.016427    -2.853
SeaSalt SALA       :           4.507904            4.515466      0.007563     0.168
SeaSalt SALAAL     :           4.507904            4.515466      0.007563     0.168
SeaSalt SALACL     :           2.481150            2.485313      0.004163     0.168
SeaSalt SALC       :         272.839025          264.549545     -8.289480    -3.038
SeaSalt SALCAL     :         272.839025          264.549545     -8.289480    -3.038
SeaSalt SALCCL     :         150.170606          145.608072     -4.562534    -3.038
```

I am curious about the changes to MEGAN biogenic emissions. I'm not sure that these are expected, and I am currently investigating the cause.

```
###################################################################################
### Emissions totals for inventory MEGAN [Tg]                                   ###
### Ref = 13.2.0_dev; Dev = 13.2.0_dev+blowingSnow                              ###
###################################################################################
                                    Ref                 Dev     Dev - Ref    % diff
MEGAN ACET         :           4.975494            4.978412      0.002918     0.059
MEGAN ACET DIRECT  :           3.769004            3.769004      0.000000     0.000
MEGAN ACET MBOX    :           0.163436            0.163436      0.000000     0.000
MEGAN ACET MONO    :           0.452801            0.452801      0.000000     0.000
MEGAN ALD2         :           1.741646            1.741453     -0.000193    -0.011
MEGAN EOH          :           1.741646            1.741453     -0.000193    -0.011
MEGAN ISOP         :          34.626235           34.663945      0.037710     0.109
MEGAN LIMO         :           0.964602            0.964677      0.000075     0.008
MEGAN MOH          :          10.605231           10.597111     -0.008121    -0.077
MEGAN MTPA         :           8.568430            8.567809     -0.000621    -0.007
MEGAN MTPO         :           3.946417            3.946165     -0.000253    -0.006
MEGAN PRPE         :           2.500340            2.502181      0.001841     0.074
MEGAN SOAP         :           1.216905            1.217482      0.000578     0.047
MEGAN SOAS         :           1.216905            1.217482      0.000578     0.047
```

Any thoughts or comments @lizziel @jaegle?

msulprizio commented:
Following up on the differences in MEGAN emissions, I ran a 1-month benchmark comparing to my internal benchmarks and got zero differences from that extension. I was able to reproduce the exact differences from the SeaSalt extension. There must be a separate issue (e.g. run setup, compiler) causing the differences in MEGAN emissions seen in the results posted above.

jaegle commented:
> Following up on the differences in MEGAN emissions, I ran a 1-month benchmark comparing to my internal benchmarks and got zero differences from that extension. I was able to reproduce the exact differences from the SeaSalt extension. There must be a separate issue (e.g. run setup, compiler) causing the differences in MEGAN emissions seen in the results posted above.

I have looked at the benchmaks and it looks good to me, with changes in SALA, SALC, BrSALA and BrSALC as expected. As this is a simulation for July, the largest changes are seen in the SH polar regions. With decreasing sea salt emissions from the oceans due to to cold temperature suppression and increased emissions over sea ice due to blowing snow sea salt emissions.

I have two quick questions:

1) did the blowing snow sea salt emissions also get added to the sea salt chloride (SALACL and SALCCL). I didn't have that in the original files as the GEOS-Chem version I was using didn't have chlorine chemistry. From the benchmark simulations it looks like it is the case, but I wanted to double check.

2) I sent a message last week about updated README and MULTI_ICE files to supersede the ones I sent a year ago, but I am not sure that this got posted on github (or I can't find it on there) and updated, so I am re-sending it below:
"I have created a new README file and updated all the multi year ice fraction files. Please upload them from

ftp.atmos.washington.edu
cd jaegle/MULTI_ICE

and use these new files instead of the ones I sent last year."

lizziel commented:
> did the blowing snow sea salt emissions also get added to the sea salt chloride (SALACL and SALCCL). I didn't have that in the original files as the GEOS-Chem version I was using didn't have chlorine chemistry. From the benchmark simulations it looks like it is the case, but I wanted to double check.

I did not make further changes to your original code submission for sea salt chloride. Should the changes be [here](https://github.com/geoschem/HEMCO/blob/051d59b391b2cad24c3ef1da96f54ad5a184ac78/src/Extensions/hcox_seasalt_mod.F90#L394-L416), adding SNOWSALA to FLUXSALACL and SNOWSALC to FLUXSALCCL, then multiplying by scale factor 0.5504?

> I sent a message last week about updated README and MULTI_ICE files to supersede the ones I sent a year ago, but I am not sure that this got posted on github (or I can't find it on there) and updated, so I am re-sending it below:
> "I have created a new README file and updated all the multi year ice fraction files. 

I don't think that message made it onto GitHub for some reason. I will download the new files to replace what we have. Thanks for bringing this up!

jaegle commented:
> I did not make further changes to your original code submission for sea salt chloride. Should the changes be [here](https://github.com/geoschem/HEMCO/blob/051d59b391b2cad24c3ef1da96f54ad5a184ac78/src/Extensions/hcox_seasalt_mod.F90#L394-L416), adding SNOWSALA to FLUXSALACL and SNOWSALC to FLUXSALCCL, then multiplying by scale factor 0.5504?

Yes, that it indeed the change that should be made. Thanks!

lizziel commented:
The updates discussed on this thread are now all set in the new PRs I created for GEOS-Chem and HEMCO. I will close this PR since it will not be merged.
